### PR TITLE
Chri7325/clang tidy 5

### PR DESCRIPTION
**Pull request by chri7325**
All unit tests pass

### CODE REVIEW

File: clang-tools-extra/clang-tidy/esri/EsriTidyModule.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+//===--- EsriTidyModule.cpp - clang-tidy ----------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
```
Comment (dkalowsk) [inline clang-tools-extra/clang-tidy/esri/EsriTidyModule.cpp]: Do we need to an any Esri specific license verbage here?

File: clang-tools-extra/clang-tidy/esri/NoImplementationInHeadersCheck.cpp [hunk 2]
```diff
@@ -0,0 +1,185 @@
+//===--- NoImplementationInHeadersCheck.cpp - clang-tidy-------------------===//
+//
+//                     The LLVM Compiler Infrastructure
```
Comment (dkalowsk) [inline clang-tools-extra/clang-tidy/esri/NoImplementationInHeadersCheck.cpp]: Do we need Esri license verbage here?

File: clang-tools-extra/clang-tidy/esri/NoImplementationInHeadersCheck.cpp [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+      return;
+    // Function templates instantiations are allowed in headers.
+    if (FD->isTemplateInstantiation())
+      return;
+
+    // rule of five defaults are prohibited in headers.
+    if (FD->isDefaulted()) {
+      diag(FD->getLocation(),
+           "rule of five function %0 defaulted in a header file; "
+           "runtimecore prohibits defaults in headers and should in source")
```
Comment (dkalowsk) [inline clang-tools-extra/clang-tidy/esri/NoImplementationInHeadersCheck.cpp]: error message is kind of broken english.  Basically missing a `be` in there.

File: clang-tools-extra/clang-tidy/esri/EsriTidyModule.cpp (hunk 1 above)
Comment (chri7325) [reply clang-tools-extra/clang-tidy/esri/EsriTidyModule.cpp]: I don't think so.  When adding clang tidy extentions, it is recommended to use their script which will auto add and generate these files with the llvm verbage.  Looking at the checks for google, there isn't any google verbage there either.  @Craig-Proctor would probably know better though.

Comment (Craig-Proctor) [reply clang-tools-extra/clang-tidy/esri/EsriTidyModule.cpp]: I think at some point we want to get this back into the main branch. We should follow the licensing they have.

File: clang-tools-extra/clang-tidy/esri/NoImplementationInHeadersCheck.cpp (hunk 3 above)
Comment (chri7325) [reply clang-tools-extra/clang-tidy/esri/NoImplementationInHeadersCheck.cpp]: Thanks!  Forgot to mention that I updated but it looks like you saw it :)

**Pull request merged**